cmd/server/main.go

```diff
@@ -7,186 +7,305 @@ import (
+	stmt, err := tx.Prepare("INSERT INTO products_table (name, description, price) VALUES (?, ?, ?)")
+	if err != nil {
+		w.WriteHeader(400)
 		return
 	}
-	prod.ID = len(products) + 1
-	products = append(products, prod)
-	json.NewEncoder(w).Encode(prod)
+	defer stmt.Close()
+
+	_, err = stmt.Exec(prod.Name, prod.Description, prod.Price)
+	if err != nil {
+		w.WriteHeader(400)
+		return
+	}
+	if err := tx.Commit(); err != nil {
+		w.WriteHeader(400)
+		return
+	}
+
+	w.WriteHeader(201)
 }
 
 func updateProduct(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
+
 	params := mux.Vars(r)
-	var prod product
-	id, err := strconv.Atoi(params["id"])
-	if err != nil {
-		log.Println("error", err)
-		w.WriteHeader(http.StatusInternalServerError)
-	}
-	for index, item := range products {
-		if id == item.ID {
-			products = append(products[:index], products[index+1:]...)
-			if err := json.NewDecoder(r.Body).Decode(&prod); err != nil {
-				log.Println("error", err)
-				w.WriteHeader(http.StatusInternalServerError)
-				return
-			}
-			products = append(products, prod)
-			json.NewEncoder(w).Encode(prod)
-			return
-		}
+	paramsConvert, err := strconv.Atoi(params["id"])
+	if paramsConvert < 0 {
+		w.WriteHeader(400)
+		return
+	}
+	if err != nil {
+		w.WriteHeader(400)
+		return
+	}
+
+	var products product
+	if err := json.NewDecoder(r.Body).Decode(&products); err != nil {
+		w.WriteHeader(400)
+		return
+	}
+	if products.Name == "" || products.Price == 0 {
+		w.WriteHeader(400)
+		return
+	}
+	tx, err := db.Begin()
+	if err != nil {
+		w.WriteHeader(500)
+		return
+	}
+	stmt, err := tx.Prepare("UPDATE products_table SET name = ?, description = ?, price = ? WHERE id = ?")
+	if err != nil {
+		w.WriteHeader(400)
+		return
+	}
+	defer stmt.Close()
+
+	res, err := stmt.Exec(products.Name, products.Description, products.Price, params["id"])
+	if err != nil {
+		w.WriteHeader(404)
+		return
+	}
+
+	if err := tx.Commit(); err != nil {
+		w.WriteHeader(400)
+		return
 	}
-	json.NewEncoder(w).Encode(products)
+	rows, err := res.RowsAffected()
+	if err != nil {
+		w.WriteHeader(500)
+		return
+	}
+
+	if rows == 0 {
+		w.WriteHeader(400)
+		return
+	}
+	w.WriteHeader(200)
 }
 
 func deleteProduct(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	params := mux.Vars(r)
-	id, err := strconv.Atoi(params["id"])
+
+	if _, err := strconv.Atoi(params["id"]); err != nil {
+		w.WriteHeader(400)
+		return
+	}
+
+	tx, err := db.Begin()
 	if err != nil {
-		log.Println("error", err)
-		w.WriteHeader(http.StatusInternalServerError)
+		w.WriteHeader(400)
+		return
 	}
-	for index, item := range products {
-		if id == item.ID {
-			products = append(products[:index], products[index+1:]...)
-			break
-		}
+
+	stmt, err := tx.Prepare("DELETE FROM products_table WHERE id = ?")
+	if err != nil {
+		w.WriteHeader(400)
+		return
+	}
+	defer stmt.Close()
+
+	res, err := stmt.Exec(params["id"])
```

Comment:
You should really ensure this is an integer value before attempting to work with it as you are.